File: src/main/java/com/condominium/management/condominium/service/CondominiumService.java
package com.condominium.management.condominium.service;

import com.condominium.management.condominium.dto.*;
import com.condominium.management.condominium.entity.Condominium;
import com.condominium.management.condominium.repository.CondominiumRepository;
import com.condominium.management.exception.ResourceNotFoundException;

import lombok.RequiredArgsConstructor;

import java.util.List;

import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CondominiumService {

    private final CondominiumRepository repository;

    public CondominiumResponseDTO create(
            CondominiumRequestDTO dto
    ) {

        Condominium condominium = Condominium.builder()
                .name(dto.name())
                .cnpj(dto.cnpj())
                .email(dto.email())
                .phone(dto.phone())
                .address(dto.address())
                .status(dto.status())
                .build();

        condominium = repository.save(condominium);

        return toResponse(condominium);
    }

    public List<CondominiumResponseDTO> findAll() {

        return repository.findAll()
                .stream()
                .map(this::toResponse)
                .toList();
    }

    public CondominiumResponseDTO findById(Long id) {

        Condominium condominium = repository.findById(id)
                .orElseThrow(() ->
                        new ResourceNotFoundException(
                                "Condomínio não encontrado"
                        )
                );

        return toResponse(condominium);
    }

    public CondominiumResponseDTO update(
            Long id,
            CondominiumRequestDTO dto
    ) {

        Condominium condominium = repository.findById(id)
                .orElseThrow(() ->
                        new ResourceNotFoundException(
                                "Condomínio não encontrado"
                        )
                );

        condominium.setName(dto.name());
        condominium.setCnpj(dto.cnpj());
        condominium.setEmail(dto.email());
        condominium.setPhone(dto.phone());
        condominium.setAddress(dto.address());
        condominium.setStatus(dto.status());

        condominium = repository.save(condominium);

        return toResponse(condominium);
    }

    public void delete(Long id) {

        if (!repository.existsById(id)) {
            throw new ResourceNotFoundException(
                    "Condomínio não encontrado"
            );
        }

        repository.deleteById(id);
    }

    private CondominiumResponseDTO toResponse(
            Condominium condominium
    ) {

        return new CondominiumResponseDTO(
                condominium.getId(),
                condominium.getName(),
                condominium.getCnpj(),
                condominium.getEmail(),
                condominium.getPhone(),
                condominium.getAddress(),
                condominium.getStatus()
        );
    }
}

File: src/services/condominiumService.ts
import api from "./api";
import type {
    Condominium,
    CondominiumRequest,
} from "../types/condominium";

const condominiumService = {

    async findAll(): Promise<Condominium[]> {
        const response = await api.get<Condominium[]>(
            "/api/condominiums"
        );

        return response.data;
    },

    async findById(id: number): Promise<Condominium> {
        const response = await api.get<Condominium>(
            `/api/condominiums/${id}`
        );

        return response.data;
    },

    async create(
        data: CondominiumRequest
    ): Promise<Condominium> {
        const response = await api.post<Condominium>(
            "/api/condominiums",
            data
        );

        return response.data;
    },

    async update(
        id: number,
        data: CondominiumRequest
    ): Promise<Condominium> {
        const response = await api.put<Condominium>(
            `/api/condominiums/${id}`,
            data
        );

        return response.data;
    },

    async delete(id: number): Promise<void> {
        await api.delete(`/api/condominiums/${id}`);
    },
};

export default condominiumService;